core/AssetManager/UIAssetCacheBuster.php

```diff
@@ -47,7 +47,7 @@ public function piwikVersionBasedCacheBuster($pluginNames = false)
                 }
             }
 
-            $cacheBuster = md5($pluginsInfo . PHP_VERSION . Version::VERSION . trim($currentGitHash ?? ''));
+            $cacheBuster = md5($pluginsInfo . PHP_VERSION . Version::VERSION . trim($currentGitHash));
```

Comment:
Have you checked all the places where you removed a `?? ''` in detail? We've added a lot of them because they were throwing warning/notices in newer PHP versions. Guess that now happens in the tests again.
The problem in many cases might be, that the parameters are defined as `string` in the doc block somewhere, while they do not have a type hint (because they might also receive other values like `null` or `false`). PHPStan in that case assumes the doc block is correct and suggests to remove `?? ''` as it's assumed to be a string always, while it isn't. In such cases the doc blocks might need to be fixed instead.